lib/logstash/outputs/elasticsearch.rb

```diff
@@ -81,6 +81,10 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # Elasticsearch with the same ID.
   config :document_id, :validate => :string, :default => nil
 
+  # A routing override to be applied to all processed events.
+  # This can be dynamic using the `%{foo}` syntax.
+  config :routing, :validate => :string, :default => nil
```

Comment:
This config variable should not have a default. But I agree is not only something for this, as it is already done with document id.

Defaults should be viable values.